### DISCUSSION TITLE
LinqToDb CLI - Any possibility to overwrite how DataContext is generated?

### DISCUSSION THREAD

**Discussion by croban** [linq2db/linq2db#3877] [Q&A]
Is there any way to change how DataLayerContext properties are generated?

This is default generation:
`public ITable<DboSomeEntity>       SomeEntity=> this.GetTable<DboSomeEntity>();`

Would like to have something like:
```
private DefaultRepository<DboSomeEntity>? _someEntityRepository;
public DefaultRepository<DboSomeEntity> SomeEntityRepository => _someEntityRepository ??= new DefaultRepository<DboSomeEntity>(this);
```

**Comment by MaceWindu** [comment]
Currently not possible

**Comment by croban** [comment]
Ah :( I need it, any possibility to create feature or to explain me where to start, i could try to find time and implement by myself if possible.

**Comment by MaceWindu** [comment]
Context is partial class, so you can have add them manually as workaround. 

There is an issue already about such kind of customizations https://github.com/linq2db/linq2db/issues/3682
